arroyo-df/src/physical.rs

```diff
@@ -156,7 +154,119 @@ pub fn window_scalar_function() -> ScalarUDF {
     )
 }
 
-impl FunctionRegistry for EmptyRegistry {
+#[derive(WrapperApi)]
+struct UdfDylibInterface {
+    run: unsafe extern "C" fn(
+        args: Vec<(FFI_ArrowArray, FFI_ArrowSchema)>,
+    ) -> (FFI_ArrowArray, FFI_ArrowSchema),
+}
+
+pub struct UdfDylib {
+    name: String,
+    signature: Signature,
+    return_type: DataType,
+    udf: Container<UdfDylibInterface>,
+}
+
+impl Debug for UdfDylib {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("UdfDylib").finish()
+    }
+}
+
+impl UdfDylib {
+    pub async fn new(name: &str, config: &DylibUdfConfig) -> Self {
+        let signature = Signature::exact(
+            config
+                .arg_types
+                .iter()
+                .map(|arg| arrow_type_to_data_type(arg))
+                .collect(),
+            Volatility::Volatile,
+        );
+
+        let artifact_url =
```

Comment:
I think the full path needs to come from the job definition and should be set at UDF compile time. Imagine that I start a pipeline, then change my artifact directory—that would break all future runs for existing jobs.

Reply:
Done. I'm now getting getting the storage provider with `StorageProvider::for_url("/")`. Is that what you had in mind?

Reply:
This is what StorageProvider::get_url is for. See the example here: https://github.com/ArroyoSystems/arroyo/blob/master/copy-artifacts/src/main.rs#L29.